src/parsers/manifest/dash/node_parsers/SegmentURL.ts

```diff
@@ -14,48 +14,48 @@
  * limitations under the License.
  */
 
-import log from "../../../../log";
-import { parseByteRange } from "./utils";
+import {
+  parseByteRange,
+  ValueParser,
+} from "./utils";
 
 export interface IParsedSegmentURL { media?: string;
                                      mediaRange?: [number, number];
                                      index?: string;
                                      indexRange?: [number, number]; }
 
 /**
- * @param {Element} root
- * @returns {Object}
+ * Parse a SegmentURL element into a SegmentURL intermediate
+ * representation.
+ * @param {Element} root - The SegmentURL root element.
+ * @returns {Array}
  */
-export default function parseSegmentURL(root : Element) : IParsedSegmentURL {
+export default function parseSegmentURL(
+  root : Element
+) : [IParsedSegmentURL, Error[]] {
   const parsedSegmentURL : IParsedSegmentURL = {};
+  const warnings : Error[] = [];
+  const parseValue = ValueParser(parsedSegmentURL, warnings);
   for (let i = 0; i < root.attributes.length; i++) {
     const attribute = root.attributes[i];
     switch (attribute.name) {
+
       case "media":
         parsedSegmentURL.media = attribute.value;
         break;
-      case "indexRange": {
-        const indexRange = parseByteRange(attribute.value);
-        if (!Array.isArray(indexRange)) {
-          log.warn(`DASH: invalid indexRange ("${attribute.value}")`);
-        } else {
-          parsedSegmentURL.indexRange = indexRange;
-        }
-      }
+
+      case "indexRange":
+        parseValue("indexRange", attribute.value, parseByteRange, "indexRange");
```

Comment:
It is too bad that `parseValue `arguments aren't clear to read.

Reply:
Yeah, you're right.

I see what I can do

Reply:
What about:

```ts
        parseValue(attribute.value, { key: "indexRange",
                                      parser: parseByteRange,
                                      mpdName: "indexRange" });
```
?

Reply:
done